addons/point_of_sale/models/pos_order.py

```diff
@@ -592,7 +592,7 @@ def _create_invoice(self, move_vals):
                             })
 
                     else:
-                        self.env['account.move.line'].with_context(skip_invoice_sync=True).create({
+                        self.env['account.move.line'].sudo().with_context(skip_invoice_sync=True).create({
```

Comment:
@vlst-odoo Changing the access right of the pos_user triggered an error here, where the user was not able to use rounding.

Reply:
Why do we need a sudo here?
Should there be a rule like `rule_invoice_pos_user` for `model_account_move_line`?

Reply:
We could indeed add a rule so that pos_user could have access to account_move_line but I don't think they really need to have access to this field. So using sudo() here avoid giving them access to it outside of the scope of creating the invoice.